### PR TITLE
feat: added Navigation docs page

### PR DESCRIPTION
**Pull request by anushasunil**
- added docs page for `Navigation`
- refactored the code with semantic HTML

image

### CODE REVIEW

File: Stylesheets/Components/navigation.css [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.demo-wrapper nav{
+    width: 100%
+}
```
Comment (dev-enforced) [inline Stylesheets/Components/navigation.css]: If this style applies only to the navigation component present in the documentation then you can shift it to the documentation.css file !!

**Review by dev-enforced — APPROVED** [review]
Liked the way you used CSS utility classes to develop the navigation component and reduced the effort on your CSS navigation file!! Good work @anushasunil 
The live preview is also looking good: https://deploy-preview-43--mondayblues.netlify.app/stylesheets/docs/navigation/docs_navigation

File: Stylesheets/Docs/navigation/docs_navigation.html [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    <link rel="stylesheet" href="/Stylesheets/Docs/codeblocks/codeblocks.css">
+    <link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.4.0/styles/base16/railscasts.min.css"
+        rel="stylesheet" />
+</head>
+
+<body>
+    <div class="navigation-page-contents">
+        <nav class="display-align-center display-justify-space-between">
+            <div class="heading display-align-center">
+                <div class="hamburger-nav clickable-image">
```
Comment (vanshitalavange) [inline Stylesheets/Docs/navigation/docs_navigation.html:21]: I like how you have created a hamburger menu of your own, instead of using a icon

**Review by vanshitalavange — APPROVED** [review]
The UI looks great! Code looks clean and structured, good use of utility classes throughout.

**Pull request merged**